internals/daemon/api_notices.go

```diff
@@ -167,5 +247,13 @@ func v1GetNotice(c *Command, r *http.Request, _ *UserState) Response {
 	if notice == nil {
 		return statusNotFound("cannot find notice with id %q", noticeID)
 	}
+	reqUid := extractRequestUid(r)
+	noticeUid := notice.User()
+	if reqUid != 0 && reqUid != noticeUid && noticeUid != -1 {
+		// Requests from non-root users may only receive notices with matching
+		// UID or UID of -1. Don't leak information about whether notice exists
+		// if user is not allowed to receive it.
+		return statusNotFound("cannot find notice with id %q", noticeID)
```

Comment:
We presumably don't want to leak whether a notice with a given ID exists if the requesting user is not permitted to receive that notice. However, assuming ID is monotonically increasing, one can discover this anyway by creating a new notice; if the new notice ID is greater, then this previous notice must presumably also exist.

Thus, this could instead return a "user not permitted" error of some sort instead of a 404. I have no strong preference either way.